src/DI/Container.php

```diff
@@ -108,4 +111,17 @@ public function tabs() {
 	public function ui() {
 		return new UIServices($this);
 	}
+
+    /**
+     * Get interface to the http service.
+     *
+     * @return GlobalHttpState
+     */
+	public function http() {
+	    return new HTTPServices(
```

Comment:
LGTM.

I think the own entry point for the WAC makes it even easier to inject the GlobalHTTPState, but I might be wrong. You could always use $DIC["http"].

The thing that lets me argue about the exposal of the GlobalHTTPState via a method on the DIC is twofold. The exposal of said methods kind of promotes a dependency antipattern (service locator). We still need that methods to make a migration easier. That is why we decided to only expose some services that are used a lot to allow your IDEs to display hints. Second thing is, that the general existence of the GlobalHTTPState is kind of an antipattern to the request -> response model implied by PSR 7. Both points aren't too crucial, though, so this is more a "but..." then a "NO!".